File: smart-contract/assembly/contracts/internals/storageKeys/tags.ts
import { stringToBytes } from '@massalabs/as-types';

export const FILE_TAG = stringToBytes('\x01FILE');
export const FILE_LOCATION_TAG = stringToBytes('\x02LOCATION');
export const CHUNK_TAG = stringToBytes('\x03CHUNK');
export const CHUNK_NB_TAG = stringToBytes('\x04CHUNK_NB');
export const FILE_METADATA_TAG = stringToBytes('\x05FM');
export const GLOBAL_METADATA_TAG = stringToBytes('\x06GM');
export const DEWEB_VERSION_TAG = stringToBytes('\xFFDEWEB_VERSION');

File: server/pkg/website/storagekeys/tags.go
package storagekeys

import "github.com/massalabs/station/pkg/convert"

const (
	FILE_TAG                   = "\x01FILE"
	FILE_LOCATION_TAG          = "\x02LOCATION"
	CHUNK_TAG                  = "\x03CHUNK"
	CHUNK_NB_TAG               = "\x04CHUNK_NB"
	FILE_METADATA_TAG          = "\x05FM"
	GLOBAL_METADATA_TAG        = "\x06GM"
	FILE_METADATA_LOCATION_TAG = "\x07FML"
	DEWEB_VERSION_TAG          = "\xFFDEWEB_VERSION"
)

// FileLocationTag is the tag for the file location as a byte array.
func FileLocationTag() []byte {
	return convert.ToBytes(FILE_LOCATION_TAG)
}

func FileTag() []byte {
	return convert.ToBytes(FILE_TAG)
}

func MetadataFileTag() []byte {
	return convert.ToBytes(FILE_METADATA_TAG)
}

func GlobalMetadataTag() []byte {
	return convert.ToBytes(GLOBAL_METADATA_TAG)
}
